Scottish-Tech-Army/Volunteer-app
# SVA-172 | Register interest in a project

- [Description](#description)
  - [Type of change](#type-of-change)
  - [How to test](#how-to-test)
- [Checklist](#checklist)

![Register interest - app screen](https://user-images.githubusercontent.com/25086222/164989538-46b2868b-7f73-497f-b464-c8d2577998dc.png)

![Slack message](https://user-images.githubusercontent.com/25086222/164989546-0658ba40-dcbc-4eac-8fb2-343ecbe2e7fb.png)

# Description

Allow a user of the app to register their interest in a project.

* The user gets to this screen by clicking on the 'Register Interest' button on a project detail screen
* This sends a request to our API
* Our API then sends a notification to the appropriate Slack channel (in production this will be initial-triage -- in testing it's initial-triage-test) for someone to follow-up with the user
* Because we haven’t implemented authentication yet, for now the user will manually enter their personal details
* We are also now more easily setup if we need to post other messages to Slack in the future via our API

Ticket: [SVA-172](https://sta2020.atlassian.net/jira/software/projects/SVA/boards/145?selectedIssue=SVA-172)

## Type of change


## How to test

To fully test this on your local machine, you'll need to:

- Add two new variables to your `/api/.env` file called `SLACK_CHANNEL_VOLUNTEER_PROJECT_INTEREST` and `SLACK_SECRET_WEBHOOK_INITIAL_TRIAGE_TEST` -- please DM me on Slack and I can send you the values for these (if you're already running the API Node server in a terminal window, you need to stop it and restart it for this change to work)
- Access the `initial-triage-test` channel on Slack so you can see the messages coming through -- I can add you to this
- If you're on a Mac, I'm not sure if you may also need to do this: go to the `/app/ios` directory in a terminal window and run `pod install`.  I've only tested on an Android machine, but this command was in the instructions for the date-picker component we're using.

# Checklist
